lib/salus/scanners/base.rb

```diff
@@ -76,11 +79,26 @@ def run!(salus_report:, required:, pass_on_raise:, reraise:)
       salus_report.add_scan_report(@report, required: required)
 
       begin
-        @report.record { run }
+        @report.record do
+          Timeout.timeout(scanner_timeout) { run }
+        end
 
         if @report.errors.any?
           pass_on_raise ? @report.pass : @report.fail
         end
+      rescue Timeout::Error
+        error_message = "Scanner #{name} timed out during execution"
+        timeout_error_data = {
+          message: error_message,
+          error_class: ScannerTimeoutError,
+          backtrace: []
+        }
+        @report.error(timeout_error_data)
+        salus_report.error(timeout_error_data)
+        bugsnag_notify(error_message)
+
+        # Propagate this error if desired
+        raise ScannerTimeoutError, timeout_error_data.message if reraise && !pass_on_raise
```

Comment:
Please delete ` && !pass_on_raise`.

Also, inside `rescue Timeout::Error` add at the breginning:
```ruby
        if @report.errors.any?
          pass_on_raise ? @report.pass : @report.fail
        end
        pass_on_raise ? @report.pass : @report.fail
```

In case you don't understand these changes, let me know.

Reply:
This would be a good spec to add:
Stub the `run` method of a scanner to let it `sleep 15`.  Then set timeout to `5` and check if salus reported timeout.

In case you don't know how to stub, please see https://www.tutorialspoint.com/rspec/rspec_stubs.htm